madmom/features/beats.py

```diff
@@ -554,6 +569,72 @@ def process(self, activations, **kwargs):
         # remove beats with negative times and return them
         return detections[np.searchsorted(detections, 0):]
 
+    def process_online(self, activations, reset=True, **kwargs):
+        """
+        Detect the beats in the given activation function for online mode.
+
+        Parameters
+        ----------
+        activations : numpy array
+            Beat activation function.
+        reset : bool, optional
+            Reset the BeatTrackingProcessor to its initial state before
+            processing.
+
+        Returns
+        -------
+        beats : numpy array
+            Detected beat positions [seconds].
+
+        """
+        # reset to initial state
+        if reset:
+            self.reset()
+        beats_ = []
+        for activation in activations:
+            # shift buffer and put new activation at end of buffer
+            buffer = self.buffer(activation)
+            # create an online interval histogram
+            histogram = self.tempo_estimator.online_interval_histogram(
+                activation, reset=False)
```

Comment:
Hmm, actually I think it's ok the way it is. The tempo_estimator gets reset inside the `reset() `method of the BeatTracker and might be called whenever `process_online()` is called.

If I want to compute a whole song then `process_online()` gets called only once with all the activations and `reset=True`. If I would use `reset=reset` here then it would reset the tempo_estimator for each activation while iterating through all activations.